File: android-app/core/core-model/src/main/java/tv/bayit/plus/core/model/CollectionModels.kt
package tv.bayit.plus.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.util.Locale

/** Item from GET /api/v1/content/collections (array response). */
@Serializable
data class CollectionListItem(
    val id: String,
    val title: String? = null,
    @SerialName("title_en") val titleEn: String? = null,
    val thumbnail: String? = null,
    val backdrop: String? = null,
    @SerialName("promo_text") val promoText: String? = null,
    @SerialName("promo_text_en") val promoTextEn: String? = null,
    @SerialName("available_movies") val availableMovies: Int = 0,
    @SerialName("total_movies") val totalMovies: Int = 0,
    @SerialName("tmdb_collection_id") val tmdbCollectionId: Int? = null,
) {
    /** Returns localized promo text based on device locale, falling back to English then Hebrew. */
    fun localizedPromoText(): String? {
        val lang = Locale.getDefault().language
        val localized: String? = if (lang == "he") promoText else promoTextEn
        return localized ?: promoTextEn ?: promoText
    }

    /** Convert to [ContentItem] for display in the VOD grid. */
    fun toContentItem(): ContentItem = ContentItem(
        id = id,
        title = title,
        description = localizedPromoText(),
        thumbnail = thumbnail,
        backdrop = backdrop,
        type = "collection",
        isCollectionParent = true,
        availableMovies = availableMovies,
        totalMovies = totalMovies,
    )
}

/** Response from GET /api/v1/content/collections/{id} */
@Serializable
data class CollectionDetail(
    val id: String,
    val title: String? = null,
    val description: String? = null,
    val thumbnail: String? = null,
    val backdrop: String? = null,
    @SerialName("available_movies") val availableMovies: Int? = null,
    @SerialName("total_movies") val totalMovies: Int? = null,
    @SerialName("promo_text") val promoText: String? = null,
    @SerialName("promo_text_en") val promoTextEn: String? = null,
    @SerialName("promo_text_es") val promoTextEs: String? = null,
    @SerialName("promo_text_fr") val promoTextFr: String? = null,
    @SerialName("promo_text_it") val promoTextIt: String? = null,
    @SerialName("promo_text_hi") val promoTextHi: String? = null,
    @SerialName("promo_text_ta") val promoTextTa: String? = null,
    @SerialName("promo_text_bn") val promoTextBn: String? = null,
    @SerialName("promo_text_ja") val promoTextJa: String? = null,
    @SerialName("promo_text_zh") val promoTextZh: String? = null,
    val movies: List<CollectionMovie> = emptyList(),
) {
    /** Returns localized promo text based on device locale, with fallback chain. */
    fun localizedPromoText(): String? {
        val lang = Locale.getDefault().language
        val localized: String? = when (lang) {
            "he" -> promoText
            "en" -> promoTextEn
            "es" -> promoTextEs
            "fr" -> promoTextFr
            "it" -> promoTextIt
            "hi" -> promoTextHi
            "ta" -> promoTextTa
            "bn" -> promoTextBn
            "ja" -> promoTextJa
            "zh" -> promoTextZh
            else -> promoTextEn
        }
        return localized ?: promoTextEn ?: promoText
    }
}

/** Movie within a collection. */
@Serializable
data class CollectionMovie(
    val id: String,
    val title: String? = null,
    val thumbnail: String? = null,
    val year: Int? = null,
    val duration: String? = null,
    @SerialName("collection_order") val collectionOrder: Int? = null,
)

File: ios-app/BayitPlusApp/Models/CollectionModels.swift
import Foundation

// MARK: - Collections

/// Item from GET /api/v1/content/collections (array response)
struct CollectionListItem: Decodable, Sendable, Identifiable {
    let id: String
    let title: String?
    let titleEn: String?
    let thumbnail: String?
    let backdrop: String?
    let promoText: String?
    let promoTextEn: String?
    let availableMovies: Int
    let totalMovies: Int
    let tmdbCollectionId: Int?

    /// Returns localized promo text based on device locale, falling back to English then Hebrew.
    var localizedPromoText: String? {
        localizedPromoText(for: Locale.current.language.languageCode?.identifier ?? "en")
    }

    /// Returns localized promo text for the given language code.
    func localizedPromoText(for lang: String) -> String? {
        let localized: String? = (lang == "he") ? promoText : promoTextEn
        return localized ?? promoTextEn ?? promoText
    }

    /// Convert to ContentItem for display in the VOD grid
    func toContentItem() -> ContentItem {
        ContentItem(
            id: id,
            title: title,
            description: localizedPromoText,
            thumbnail: thumbnail,
            backdrop: backdrop,
            duration: nil,
            year: nil,
            category: nil,
            categorySlug: nil,
            categoryNameKey: nil,
            categoryNameEn: nil,
            categoryNameEs: nil,
            type: "collection",
            totalEpisodes: nil,
            availableSubtitleLanguages: nil,
            hasSubtitles: nil,
            author: nil,
            narrator: nil,
            isCollectionParent: true,
            availableMovies: availableMovies,
            totalMovies: totalMovies,
            contentRating: nil,
            genre: nil,
            genreIds: nil
        )
    }
}

/// Response from GET /api/v1/content/collections/{id}
struct CollectionDetail: Decodable, Sendable, Identifiable {
    let id: String
    let title: String?
    let titleEn: String?
    let description: String?
    let descriptionEn: String?
    let thumbnail: String?
    let backdrop: String?
    let availableMovies: Int?
    let totalMovies: Int?
    let tmdbCollectionId: Int?
    let trailerStreamUrl: String?
    let promoText: String?
    let promoTextEn: String?
    let promoTextEs: String?
    let promoTextFr: String?
    let promoTextIt: String?
    let promoTextHi: String?
    let promoTextTa: String?
    let promoTextBn: String?
    let promoTextJa: String?
    let promoTextZh: String?
    let movies: [CollectionMovie]?

    /// Returns localized title for the given language code from Bayit+ localization.
    func localizedTitle(for lang: String) -> String? {
        switch lang {
        case "en":
            return titleEn ?? title
        default:
            return title ?? titleEn
        }
    }

    /// Returns localized description for the given language code.
    func localizedDescription(for lang: String) -> String? {
        switch lang {
        case "en":
            return descriptionEn ?? description
        default:
            return description ?? descriptionEn
        }
    }

    /// Returns localized promo text based on device locale, falling back to English then Hebrew.
    var localizedPromoText: String? {
        localizedPromoText(for: Locale.current.language.languageCode?.identifier ?? "en")
    }

    /// Returns localized promo text for the given language code.
    func localizedPromoText(for lang: String) -> String? {
        let localized: String? = switch lang {
        case "he": promoText
        case "en": promoTextEn
        case "es": promoTextEs
        case "fr": promoTextFr
        case "it": promoTextIt
        case "hi": promoTextHi
        case "ta": promoTextTa
        case "bn": promoTextBn
        case "ja": promoTextJa
        case "zh": promoTextZh
        default: promoTextEn
        }
        return localized ?? promoTextEn ?? promoText
    }

    /// Average rating across movies that have ratings.
    var averageRating: String? {
        guard let movies = movies else { return nil }
        let ratings = movies.compactMap { $0.rating?.value }.compactMap { Double($0) }
        guard !ratings.isEmpty else { return nil }
        let avg = ratings.reduce(0, +) / Double(ratings.count)
        return String(format: "%.1f", avg)
    }
}

/// Movie within a collection
struct CollectionMovie: Decodable, Sendable, Identifiable {
    let id: String
    let title: String?
    let titleEn: String?
    let thumbnail: String?
    let year: Int?
    let duration: String?
    let collectionOrder: Int?
    let rating: FlexibleRating?
    let streamUrl: String?

    /// Returns localized title for the given language code.
    func localizedTitle(for lang: String) -> String? {
        switch lang {
        case "en":
            return titleEn ?? title
        default:
            return title ?? titleEn
        }
    }
}
